lsp/src/main/scala/org/ensime/lsp/rpc/RpcFormats.scala

```diff
@@ -0,0 +1,176 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.lsp.rpc
+
+import spray.json._
+import shapeless._
+import messages._
+import JsonRpcMessages._
+import org.ensime.lsp.JsonUtils
+import shapeless.tag.@@
+
+import scala.util.{ Failure, Success, Try }
+
+object JsInnerFormats {
```

Comment:
I don't understand why this is needed. I'll try deleting it and see what fails to compile.

Reply:
ok, I understand why you did this. Sorry, I could have saved you the effort of doing this overly complex workaround. The solution is the same thing you did for `either`...  look at the comments in `JerkyFormats` to understand what the problem is and the workaround.

stalactite / scalaz-deriving solves this trivially. In fact, refactoring our json formatter to use export-hook style solves it too. I meant to do that before you started but I had no time :disappointed:

Reply:
i.e. https://github.com/ensime/ensime-server/issues/1823

Reply:
the fact that the formatters are `implicit object` instead of `implicit val` could be the core problem, I may just fix that now.